lib/search/index.ts

```diff
@@ -203,7 +203,7 @@ export const middleware: S.Middleware = (store) => {
 
       if (
         searchTerms.length > 0 &&
-        !searchTerms.every((term) => note.content.includes(term))
+        !searchTerms.every((term) => note.content.match(RegExp(term, 'gi')))
```

Comment:
instead of creating a new RegExp on every search term, which has proven to be quite slow in the past, let's transform the term itself. the content in the search index is [already transformed](https://github.com/Automattic/simplenote-electron/blob/1baaf0aa01609b4e55a05df956f7b206b93d9af3/lib/search/index.ts#L33).

```ts
!searchTerms.every((term) => note.content.includes(term.toLocaleLowerCase()))
```